src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,555 @@
+        require(operatorId == pubkey.hashG1Point(), "BLSRegistryCoordinatorWithIndices._deregisterOperatorWithCoordinator: operatorId does not match pubkey hash");
+
+        // get the quorumNumbers of the operator
+        uint256 quorumsToRemoveBitmap = BitmapUtils.orderedBytesArrayToBitmap(quorumNumbers);
+        require(quorumsToRemoveBitmap <= MiddlewareUtils.MAX_QUORUM_BITMAP, "BLSRegistryCoordinatorWithIndices._deregisterOperatorWithCoordinator: quorumsToRemoveBitmap exceeds of max bitmap size");
+        uint256 operatorQuorumBitmapHistoryLengthMinusOne = _operatorIdToQuorumBitmapHistory[operatorId].length - 1;
+        uint192 quorumBitmapBeforeUpdate = _operatorIdToQuorumBitmapHistory[operatorId][operatorQuorumBitmapHistoryLengthMinusOne].quorumBitmap;
+        // check that the quorumNumbers of the operator matches the quorumNumbers passed in
+        require(
+            quorumBitmapBeforeUpdate & quorumsToRemoveBitmap == quorumsToRemoveBitmap,
+            "BLSRegistryCoordinatorWithIndices._deregisterOperatorWithCoordinator: cannot deregister operator for quorums that it is not a part of"
+        );
+        // check if the operator is completely deregistering
+        bool completeDeregistration = quorumBitmapBeforeUpdate == quorumsToRemoveBitmap;
+        
+        // deregister the operator from the BLSPubkeyRegistry
+        blsPubkeyRegistry.deregisterOperator(operator, quorumNumbers, pubkey);
+
+        // deregister the operator from the StakeRegistry
+        stakeRegistry.deregisterOperator(operatorId, quorumNumbers);
+
+        // deregister the operator from the IndexRegistry
+        indexRegistry.deregisterOperator(operatorId, quorumNumbers, operatorIdsToSwap);
+
+        // set the toBlockNumber of the operator's quorum bitmap update
+        _operatorIdToQuorumBitmapHistory[operatorId][operatorQuorumBitmapHistoryLengthMinusOne].nextUpdateBlockNumber = uint32(block.number);
+        
+        // if it is not a complete deregistration, add a new quorum bitmap update
+        if (!completeDeregistration) {
+            _operatorIdToQuorumBitmapHistory[operatorId].push(QuorumBitmapUpdate({
+                updateBlockNumber: uint32(block.number),
+                nextUpdateBlockNumber: 0,
+                quorumBitmap: quorumBitmapBeforeUpdate & ~uint192(quorumsToRemoveBitmap) // this removes the quorumsToRemoveBitmap from the quorumBitmapBeforeUpdate
+            }));
+        } else {
+            // @notice Registrant must continue to serve until the latest block at which an active task expires. this info is used in challenges
+            uint32 latestServeUntilBlock = serviceManager.latestServeUntilBlock();
+
+            // record a stake update unbonding the operator after `latestServeUntilBlock`
+            // serviceManager.recordLastStakeUpdateAndRevokeSlashingAbility(operator, latestServeUntilBlock);
+            // set the status of the operator to DEREGISTERED
```

Comment:
this variable is unused due to the other commented-out lines. should presumably be commented out as well.